instrumentation/servlet-2.4/src/main/java/com/nr/instrumentation/servlet24/ServletHelper.java

```diff
@@ -184,7 +184,9 @@ private static boolean setTxNameUsingServletName(ServletConfig servletConfig, Se
             return false;
         }
         txName = normalizeServletName(txName, servlet);
-        NewRelic.getAgent().getLogger().log(Level.FINER, "Setting transaction name using servlet name: {0}", txName);
+        NewRelic.getAgent().getLogger().log(Level.FINEST, "Servlet24/ServletHelper::setTxNameUsingServletName: calling transaction.setTransactionName with " +
```

Comment:
Having the + to concatenate the strings will force the compiler to create a StringBuilder everytime this line is executed, even if it is not going to be logged.

I recommend either saving the String as a constant in the class, so the StringBuilder is only ever created once, or wrapping the log statement with a check to see if this is loggable.
```
if (NewRelic.getAgent().getLogger().isLoggable(Level.FINEST)) {
  // log statement
}
```

Pretty much all the other log statements could benefit from one of these. Either because of the +, or because of the call to String.join.